pkg/pipeline/decode/decode_protobuf.go

```diff
@@ -66,6 +66,7 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 		"DnsResponseTimeMs":      flow.TimeDnsRsp.AsTime().UnixMilli(),
 		"DnsId":                  flow.GetDnsId(),
 		"DnsFlags":               flow.GetDnsFlags(),
+		"DnsFlagsResponseCode":   dnsRcodeToStr(flow.GetDnsFlags() & 0xF),
```

Comment:
Would it be possible to append the field only when a value is provided ?

It would save storage

Reply:
created `Append` method and conditionally add dns or tcpdrop when they happens